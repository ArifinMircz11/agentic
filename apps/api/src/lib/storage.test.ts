import { describe, expect, it } from 'vitest'

import {
  deleteStorageObject,
  getStorageObject,
  putStorageObject
} from './storage'

describe('Storage', () => {
  it('putObject, getObject, deleteObject', async () => {
    if (!process.env.ACCESS_KEY_ID) {
      // TODO: ignore on CI
      expect(true).toEqual(true)
      return
    }

    await putStorageObject('test.txt', 'hello world', {
      ContentType: 'text/plain'
    })

    const obj = await getStorageObject('test.txt')
    expect(obj.ContentType).toEqual('text/plain')

    const body = await obj.Body?.transformToString()
    expect(body).toEqual('hello world')

    const res = await deleteStorageObject('test.txt')
    expect(res.$metadata.httpStatusCode).toEqual(204)
  })
})
