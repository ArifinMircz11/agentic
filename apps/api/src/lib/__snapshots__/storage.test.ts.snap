// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Storage > uploadFileUrlToStorage data-uri 1`] = `"https://storage.agentic.so/@dev/test/ef4238fba78887e0974cd48809a66e284cdd78ce92d6b2d485c25a552fb39631.svg"`;

exports[`Storage > uploadFileUrlToStorage data-uri 2 1`] = `"https://storage.agentic.so/@dev/test/efbc1f0409b730d93b01c918be9e024bf7777801cfd252221c39e36c08c1b4fb"`;

exports[`Storage > uploadFileUrlToStorage url 1`] = `"https://storage.agentic.so/@dev/test/6da6ef895be2a42606b99e5e3b9c25687c92c81986a9718e07f32e574d41cf7a.svg"`;
